### DISCUSSION TITLE
Its not searching the contains and postfix string.

### DISCUSSION THREAD

**Discussion by atindersingh67** [appbaseio/reactivesearch#1835] [Q&A] [answered]
I am using reactive search . Its searching the text with prefix only, but its not search text with postfix and contains.
For eg. i have description filed with the data 'product1 BN2445 with red color' if search for 'prod' , 'red' or 'color' its giving me results, 
But if i search with '2445' its not giving me result, 

My index creation code is below, 
`POST /product2/product
{
  
  "settings": {
        	"index": {
            	"analysis" : {
    "analyzer":{
        "autosuggest_analyzer": {
            "filter": [
                "lowercase",
                "asciifolding",
                "autosuggest_filter"
            ],
            "tokenizer": "standard",
            "type": "custom"
        },
        "ngram_analyzer": {
            "filter": [
                "lowercase",
                "asciifolding",
                "ngram_filter"
            ],
            "tokenizer": "standard",
            "type": "custom"
        }
    },
    "filter": {
        "autosuggest_filter": {
            "max_gram": "7",
            "min_gram": "1",
            "token_chars": [
                "letter",
                "digit",
                "punctuation",
                "symbol"
            ],
            "type": "edge_ngram"
        },
        "ngram_filter": {
            "max_gram": "9",
            "min_gram": "2",
            "token_chars": [
                "letter",
                "digit",
                "punctuation",
                "symbol"
            ],
            "type": "ngram"
        }
    }
  },
  "max_ngram_diff":"50"
        	}
    	},
"mappings" : {

        "properties" : {
          "@timestamp" : {
            "type" : "date"
          },
          "@version" : {
            "type" : "text",
            "fields" : {
              "keyword" : {
                "type" : "keyword",
                "ignore_above" : 256
              }
            }
          },
        
          "standardiseddescription" : {
            "type" : "text",
             "fields" : {
              "keyword" : {
                "type" : "keyword",
                "ignore_above" : 256
              }
            }
          }
        }
      }
}`

And my data search component is below 

` <DataSearch
                        componentId="result"
                        dataField = {[
                           "standardiseddescription"
                        ]}
                        fuzziness = {'AUTO'}
                        size = {3}
                        autosuggest={true}
                        queryFormat="and"
                        style = {{
                            marginBottom: 20
                        }}
                        URLParams={true}
                    />`

I have already spend 3 days on this. Thanks in advance

**Comment by mohdashraf010897** [comment, marked as answer]
@atindersingh67 

You're doing an infix search over here, for doing this effectively, you need to use a n-gram analyzer.

```
 "standardiseddescription" : {
        "type" : "text",
         "fields" : {
          "keyword" : {
            "type" : "keyword",
            "ignore_above" : 256
          },
          "infix-search": {
            "analyzer": "ngram_analyzer"
          }
        }
      }
```

Now change the data field in your search to `standardiseddescription.infix-search`.
